odoo/tests/common.py

```diff
@@ -1153,6 +1153,16 @@ def navigate_to(self, url, wait_stop=False):
             self._websocket_wait_event('Page.frameStoppedLoading', params={'frameId': frame_id})
 
     def clear(self):
+        clear_service_workers = """
+        if ('serviceWorker' in navigator) {
+            navigator.serviceWorker.getRegistrations().then(
+                (registrations) => {registrations.map(r => r.unregister())}
+            )
+        }
+        """
+        cl_id = self._websocket_send('Runtime.evaluate', params={'expression': clear_service_workers})
+        self._websocket_wait_id(cl_id)
```

Comment:
`clear_service_workers` snippet use a Promise (from `getRegistrations()`) but doesn't wait for it (at least in JS).

Is it normal? Is there something on the Python side that waits ?

Technically speaking, the snippet will finish its execution immediately even if the `unregister()` call(s) are not completed. In our case, that shouldn't long but still...

Reply:
Maybe a detail but you can use a `const` instead of a `let` as this variable isn't reassigned.

Reply:
Also, about the `await`, I guess that as this snippet is executed in the global context, so it works even outside an `async` ?

Reply:
@pparidans Indeed for const :) About await it worked well, but let's try the option `awaitPromise`

Reply:
Actually, found the release notes I was thinking about :wink: 
https://developers.google.com/web/updates/2017/08/devtools-release-notes#await

Top level await is supported only in the DevTools (which we use in this case), so it's fine using it here.

Reply:
Not sure about the `awaitPromise` as your expression doesn't return anything...

Reply:
Yes, you might be right, thanks

Reply:
@pparidans Mhé, awaiting like this cause sometimes an issue -_-
See https://runbot.odoo.com/runbot/build/5249178

So I think I'll comeback to my original fix which worked all the times...

Reply:
Is it really the "good fix" to not wait the promise... :thinking:

Reply:
@pparidans It seems that waiting for it do not always fix the problem... But I absolutely don't know why...
Keeping the old code and adding the option "awaitPromise" also work if you really want to wait the promise.

Reply:
I really want to wait for it :wink: 

LGTM :+1: